btree/btree.c

```diff
@@ -3026,8 +3088,19 @@ static int64_t btree_put_tick(struct m0_sm_op *smop)
 			lev->l_node = oi->i_nop.no_node;
 			node_slot.s_node = oi->i_nop.no_node;
 			lev->l_seq = lev->l_node->n_seq;
-			/* Verify node footer */
-			if (!node_verify(lev->l_node)) {
+
+			/**
+			 * Header verification is required to determine that
+			 * the node(lev->l_node) which is pointed by current
+			 * thread is not freed by any other thread till current
+			 * thread reaches NEXTDOWN phase.
+			 *
+			 * Footer verification is required to detrmine that no
+			 * other thread which has lock is working on the same
+			 * node(lev->l_node) which is pointed by current thread.
+			 */
+			if (!node_header_verify(lev->l_node) ||
+			    !node_footer_verify(lev->l_node)) {
```

Comment:
This will still not protect the data structures and cause crash at times. We need to think about this a bit more.

Reply:
okay